Enrich 5.4.0 (#1269)
snowplow/documentation: docs/pipeline/enrichments/available-enrichments/custom-javascript-enrichment/writing/index.md

:::note

You might be tempted to update derived entities in a similar way by using `event.setDerived_contexts()`. However, this is not supported (the function exists, but has no effect). Instead, refer to the [Erasing derived contexts](#erasing-derived-contexts) section.

:::

## Erasing derived contexts

Starting with Enrich 5.4.0, it is possible to erase derived contexts.

This feature can be used to update existing derived contexts as well. The way to do that is shown in the below example:

```js
function process(event) {
  const derived = JSON.parse(event.getDerived_contexts())

  // erase the existing contexts from the event
  event.eraseDerived_contexts()

  // modify the contexts

>   derived.data[0].data.field = "test"

Reviewer: ```suggestion
  for (const entity of derived.data) {
    if (entity.schema === ...) {
      // update a field inside
      entity.data.myField = entity.data.myField + 1;
    }
  }
```

  }
}
```

> ## Erasing derived contexts

Reviewer: Could you move this to right after line 242 (which talks about modifying entities/contexts)? Also there is a note there about `setDerived_contexts` not being a thing. Could you update that note to refer to this section as the alternative?

Starting with Enrich 5.4.0, it is possible to erase derived contexts.

> It can be used to update existing derived contexts as well. The way to do that is shown in the below example:

Reviewer: ```suggestion
This feature can be used to update existing derived contexts as well. The way to do that is shown in the below example:
```

```js
function process(event) {
  const derived = JSON.parse(event.getDerived_contexts());

>   event.eraseDerived_contexts();

Reviewer: ```suggestion
  // erase the existing contexts from the event
  event.eraseDerived_contexts();
```

>   derived.data[0].data.deviceBrand = "test1";

Reviewer: ```suggestion
  // modify the contexts
  derived.data[0].data.deviceBrand = "test1";
```

>   return derived.data;

Reviewer: ```suggestion
  // returned the updated array of derived contexts, which will replace the original one
  return derived.data;
```
